### PR TITLE
Coord cleanup

### PR DESCRIPTION
**Pull request by SJuliez**
Mostly cleans up some unused stuff from Coords and moves together all the getHexDonut variations into a single method in Coords. Also moves the isOnBoardEdge check from Coords to Board. No changes in function intended.

### CODE REVIEW

**Comment by sixlettervariables** [comment]
Looks like these changes may have broken some of the tests.

File: megamek/src/megamek/common/Coords.java [hunk 1]
```diff
@@ -350,26 +301,15 @@ public boolean equals(Object object) {
...
-    /**
-     * Get the hash code for these coords.
-     * 
-     * @return The <code>int</code> hash code for these coords.
-     */
+    /** Returns the hash code for these coords. */
     @Override
     public int hashCode() {
+        if (hash == 0) {
+            hash = (HashCodeUtil.hash1(x + 1337) ^ HashCodeUtil.hash1(y + 97331)) & 0x7FFFFFFF;
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/Coords.java:311]: Why not `Objects.hash(x, y)`?

Comment (SJuliez) [reply megamek/src/megamek/common/Coords.java]: Akjosch added HashCodeUtil. It has the comment:
Specialized hash code implementations suitable for PRNG initialisation
I have no idea why. I just left it as it was.

Comment (SJuliez) [reply megamek/src/megamek/common/Coords.java]: Changed it now.

**Comment by Windchild292** [comment]
This has merge conflicts @SJuliez

**Review by sixlettervariables — APPROVED** [review]

**Pull request merged**